File: summer/WebContent/js/system/componentmodel/IEditableCollectionView.js
/**
 * IEditableCollectionView
 */

define(["dojo/_base/declare", "system/Type"], function(declare, Type){
	var IEditableCollectionView = declare("IEditableCollectionView", null,{
//		object 
		AddNew:function(){},
//		void 
		CommitNew:function(){},
//		void 
		CancelNew:function(){},
//		void 
		RemoveAt:function(/*int*/ index){},
//		void 
		Remove:function(/*object*/ item){},
//		void 
		EditItem:function(/*object*/ item){},
//		void 
		CommitEdit:function(){},
//		void 
		CancelEdit:function(){},
	});
	
	Object.defineProperties(IEditableCollectionView.prototype,{
		  
//		NewItemPlaceholderPosition 
		NewItemPlaceholderPosition:
		{
			get:function(){},
			set:function(value){}
		},
//		bool 
		CanAddNew:
		{
			get:function(){}
		},
//		bool 
		IsAddingNew:
		{
			get:function(){}
		},
//		object 
		CurrentAddItem:
		{
			get:function(){}
		},
//		bool 
		CanRemove:
		{
			get:function(){}
		},
//		bool 
		CanCancelEdit:
		{
			get:function(){}
		},
//		bool 
		IsEditingItem:
		{
			get:function(){}
		},
//		object 
		CurrentEditItem:
		{
			get:function(){}
		}
	});
	
	IEditableCollectionView.Type = new Type("IEditableCollectionView", IEditableCollectionView, [Object.Type]);
	return IEditableCollectionView;
});

File: org.summer.view.widget/src/org/summer/view/widget/model/IEditableCollectionView.java
package org.summer.view.widget.model;
public interface IEditableCollectionView {
		void EditItem (Object item);
		Object AddNew ();

		void CancelEdit ();
		void CancelNew ();

		void CommitEdit ();
		void CommitNew ();

		void Remove (Object item);
		void RemoveAt (int index);

		Object CurrentAddItem { get; }
		Object CurrentEditItem { get; }

		boolean CanAddNew { get; }
		boolean CanCancelEdit { get; }
		boolean CanRemove { get; }

		boolean IsAddingNew { get; }
		boolean IsEditingItem { get; }

		NewItemPlaceholderPosition NewItemPlaceholderPosition { get; set; }
	}